src/libraries/System.Private.Xml/tests/XmlReaderLib/TestFiles.cs

```diff
@@ -759,68 +761,67 @@ public static void CreateBinHexTestFile(string strFileName)
             w.WriteElementString("ElemErr", "a&A2A3");
 
             w.WriteEndElement();
-            w.Flush();
+
             FilePathUtil.addStream(strFileName, mems);
         }
 
         public static void CreateBigElementTestFile(string strFileName)
         {
-            Stream s = new MemoryStream();
-            TextWriter tw = new StreamWriter(s);
+            var s = new MemoryStream();
+            using var tw = new StreamWriter(s, encoding:null, bufferSize:-1, leaveOpen:true);
 
             string str = new string('Z', (1 << 20) - 1);
             tw.WriteLine("<Root>");
             tw.Write("<");
             tw.Write(str);
             tw.WriteLine("X />");
-            tw.Flush();
 
             tw.Write("<");
             tw.Write(str);
             tw.WriteLine("Y />");
             tw.WriteLine("</Root>");
 
-            tw.Flush();
             FilePathUtil.addStream(strFileName, s);
         }
         public static void CreateXSLTStyleSheetWCopyTestFile(string strFileName)
         {
-            Stream s = new MemoryStream();
-            TextWriter tw = new StreamWriter(s);
+            var s = new MemoryStream();
+            using var tw = new StreamWriter(s, encoding:null, bufferSize:-1, leaveOpen:true);
+
             tw.WriteLine("<xsl:stylesheet version=\"1.0\" xmlns:xsl=\"http://www.w3.org/1999/XSL/Transform\">");
             tw.WriteLine("<xsl:template match=\"/\">");
             tw.WriteLine("<xsl:copy-of select=\"/\" />");
             tw.WriteLine("</xsl:template>");
             tw.WriteLine("</xsl:stylesheet>");
-            tw.Flush();
+
             FilePathUtil.addStream(strFileName, s);
         }
 
         public static void CreateConstructorTestFile(string strFileName)
         {
-            Stream s = new MemoryStream();
-            TextWriter tw = new StreamWriter(s);
+            var s = new MemoryStream();
```

Comment:
Would it be better to write

```suggestion
            MemoryStream s = new();
```

rather than 

```cs
var s = new MemoryStream();
```

which translates to

```cs
MemoryStream? s = new MemoryStream();
```

?

Reply:
Why?

Reply:
There are two things: 

1) Is it true that `var s = new MemoryStream();` is equivalent with `MemoryStream? s = new MemoryStream();`?
2) If 1) is true, then this PR changes `s` to be nullable variable. But is it for a good reason?

Maybe I'm wrong, I just want to understand this.

Reply:
We don't have nullable reference types enabled in our test projects.  But even if we did, the only thing it would affect is the ability to store null into the local, which either the test would want to do in which case it's desirable or the test won't do in which case it doesn't matter.  As a local it doesn't affect any code outside of the method.

Reply:
Relatedly, I am not sure we have a preference between `var foo = new Foo(..)`, `Foo foo = new(..)`, and `Foo foo = new Foo(..)`.

Reply:
It's true there is no real difference here. 

For projects with `<nullable>` enabled, i like to make variables non-nullable by default. But it seems people generally like `var foo = new Foo(...)` better. This makes me sad because it leads to cases where it's unclear what an author intended for the variable to be allowed (can it be null or not).